plugins/woocommerce/includes/class-wc-ajax.php

```diff
@@ -3065,13 +3075,21 @@ public static function shipping_zone_add_method() {
 
 		$zone_id = wc_clean( wp_unslash( $_POST['zone_id'] ) );
 		$zone    = new WC_Shipping_Zone( $zone_id );
+		if ( '' === $zone_id ) {
+			/**
+			 * Notified that a non-option setting has been added.
+			 *
+			 * @since 7.8.0
+			 */
+			do_action( 'woocommerce_add_non_option_setting', array( 'id' => 'shipping_zone' ) );
+		}
 		/**
-		 * Notify that a non-option setting has been updated.
+		 * Notify that a non-option setting has been added.
 		 *
 		 * @since 7.8.0
 		 */
 		do_action(
-			'woocommerce_update_non_option_setting',
+			'woocommerce_add_non_option_setting',
```

Comment:
The fact that we are changing this to `add` instead of `update` makes me think we should revert back to your naming scheme in the previous implementation and make it shipping zone specific -> `woocommerce_add_shipping_zone_method` although it looks like we are already firing a similar action -> `woocommerce_shipping_zone_method_added` as part of the `add_shipping_method` function.

I would stick with `woocommerce_update_non_option_setting` though to follow suite with the original thought behind this. And just tag on action type as a property:
```
array(
  'id' => 'zone_method',
  'action' => 'add'
)
```